yearn/contracts/ConvexStrategy.sol

```diff
@@ -606,7 +617,34 @@ contract ConvexStrategy is BaseStrategy {
         override
         returns (uint256)
     {
-        // TODO: Create an accurate price oracle.
-        return amtInWei;
+        address[] memory path = new address[](2);
+        path[0] = weth;
+        path[1] = wbtc;
+
+        // As of writing this contract, there's no pool available that trades
+        // an underlying token with ETH. To overcome this, the ETH amount
+        // denominated in WEI should be converted into an amount denominated
+        // in one of the tokens accepted by the tBTC v2 Curve pool using Uniswap.
+        // The wBTC token was chosen arbitrarily since it is already used in this
+        // contract for other operations on Uniswap.
+        // amounts[0] -> ETH in wei
+        // amounts[1] -> wBTC
+        uint256[] memory amounts = IUniswapV2Router(uniswap).getAmountsOut(
+            amtInWei,
+            path
+        );
+
+        // Use the amount denominated in wBTC to calculate the amount of LP token
+        // (vault's underlying token) that could be obtained if that wBTC amount
+        // was deposited in the Curve pool that has tBTC v2 in it. This way we
+        // obtain an estimated value of the original WEI amount represented in
+        // the vault's underlying token.
+        // TODO: When the new curve pool with tBTC v2 is deployed, verify that
+        // the index of wBTC (amounts[1]) in the array is correct.
+        return
+            ICurvePool(tbtcCurvePoolDepositor).calc_token_amount(
+                [amounts[1], 0, 0, 0],
```

Comment:
Shouldn't we pass `amounts[1]` at index `2` like we do in `CurveVoterProxyStrategy`?

Reply:
Yes! I forgot to change it here, done in 1811272